File: 01/js/first.js
/*
 * Run using deno
 *
 * $ cat input.txt | time deno run ./js/first.js
 */

import { readLines } from "https://deno.land/std/io/mod.ts";
let t=0,z
for await (const s of readLines(Deno.stdin)) {
	if (!s) break;
	z=s.split``.filter(n=>!Number.isNaN(+n)),t+=+`${z[0]}${z[z.length-1]}`
}
console.log(t)

File: 01/cpp/first.cpp
/*
 * Compile using gcc
 * 
 * $ g++ ./first.cpp -o first.out
 * 
 * Run directly
 * 
 * $ cat input.txt | time ./cpp/first.out
 */

#include <iostream>
#include <string>

int main() {
    std::string s;
    int n;
    while (true) {
        std::getline(std::cin, s);
        for (int i = 0; i < s.size(); i++) {
            char c = s.at(i);
            if ('0' <= c && c <= '9') {
                n += (c - '0') * 10;
                goto second;
            }
        }
        break;
    second:
        for (int i = s.size(); i > 0; i--) {
            char c = s.at(i - 1);
            if ('0' <= c && c <= '9') {
                n += c - '0';
                s.clear();
                goto next;
            }
        }
        break;
    next:
        continue;
    }
    std::cout << n << std::endl;
}